templates/skeleton/app/components/Footer.tsx

```diff
@@ -1,29 +1,45 @@
-import {NavLink} from '@remix-run/react';
-import type {FooterQuery, HeaderQuery} from 'storefrontapi.generated';
-import {useRootLoaderData} from '~/lib/root-data';
+import {Await, NavLink} from '@remix-run/react';
+import {Suspense} from 'react';
+import type {RootLoaderData} from '~/root';
+
+type FooterProps = Pick<
+  RootLoaderData,
+  'footer' | 'header' | 'publicStoreDomain'
+>;
```

Comment:
I feel this might not be secure because if the root loader changes, this props type is changed automatically. It would be better to rely on the generated type of the query, I think?

Maybe we should move the queries and fragments related to Header and Footer to these components? That way they control the data, and we can just rely on the generated types for the props.

And probably just use `publicStoreDomain: string` to make it strict, instead of getting it from a type?

---

This would also be useful to reduce the code in `root.tsx` for diff examples.